### PR TITLE
Remove requirement for certificate authority and certificate authority data in kubeconfig.

### PR DESCRIPTION
**Pull request by twitchax**
This PR addresses #220.

Basically, we remove the requirement upon validation (and the impacted test) for certificate authority data in the `kubeconfig` file.  The `kubeconfig` will work just fine in `kubectl` without this data if the CA is well-known (e.g., LetsEncrypt).  Therefore, this PR removes that validation step, thereby pushing any failures due to unknown CA issues to the time of negotiation.

### CODE REVIEW

**Comment by twitchax** [comment]
I already signed an agreement and linked to GitHub.  Maybe it needs time to propagate?

**Comment by brendandburns** [comment]
@twitchax your commits have a different email address, I think, see:
https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

**Comment by twitchax** [comment]
@brendandburns, I see.  Fixed and force pushed.

**Comment by twitchax** [comment]
@itowlson, correct on the confusing title.  Fixed.

File: src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs [hunk 1]
```diff
@@ -155,45 +155,32 @@ private void SetClusterDetails(K8SConfiguration k8SConfig, Context activeContext
 
             if (clusterDetails?.ClusterEndpoint == null)
             {
-                throw new KubeConfigException($"Cluster not found for context {activeContext} in kubeconfig");
+                throw new KubeConfigException($"Cluster not found for context `{activeContext}` in kubeconfig");
```
[the remark this answers is not in the document]
Comment (twitchax) [reply src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs]: The backticks are personal preference, but I do think that specific names of things referenced in an error message should be delimited in one manner or another.  Switch to single quotes or remove?

**Comment by twitchax** [comment]
@itowlson, ping. :)

**Comment by brendandburns** [comment]
This looks good to me. @itowlson are you ok with the tick-marks?

**Comment by itowlson** [comment]
I don't feel strongly enough about the use of backticks to think it's worth holing this PR for. I'd like to have a convention for how we delimit names but it can wait.

/lgtm

**Comment by twitchax** [comment]
@itowlson, I do not mind updating the PR for single quotes or removing them.  Just let me know.

**Pull request merged**